scripts/pre_commit_linter.py

```diff
@@ -0,0 +1,153 @@
+
+        python scripts/pre_commit_linter.py
+ Note that the root folder MUST be named 'oppia'.
+ """
+
+__author__ = "Barnabas Makonda(barnabasmakonda@gmail.com)"
+
+import os
+import subprocess
+import sys
+
+
+def _get_list_of_current_changed():
+    """This function combines all changed files staged
+    and not staged
+    Returns:
+        a list of files about to be commited. """
+    changed_and_staged_files = subprocess.check_output(['git', 'diff', '--name-only'])
+    changed_and_staged_files += subprocess.check_output(['git', 'diff', '--cached',
+                                    '--name-only', '--diff-filter=ACM'])
+    files = []
+    if changed_and_staged_files:
+        for changed_and_staged_file in changed_and_staged_files.splitlines():
+            files.append(changed_and_staged_file)
+    return files
+
+
+def _is_javascript_file(filename):
+    """Check if the input file looks like a javascript
+    Args:
+        filename(String): Name of the files to be checked
+            if is javascript or not
+    Returns:
+        bool: True if the filename ends in ".js" and false otherwise.
+    """
+    if filename.endswith('.js'):
+        return True
+    else:
+        return False
+
+
+def _check_repo(path_to_jscs):
+    """This function checks all changed files in the repo and
+    choose only javascript files(those which ends with .js)
+    to be linted and the output error is displayed in console
+    Args:
+        path_to_jscs(String): path to the jscs bin direcory
+        auto_fix(Optional[bool]): if true node-jscs auto fix all
+           errors that it can and return remaining errors.
+    Raises:
+        valueError:if file was staged and then deleted
+    Returns:
+        errors: A list of errors obtained
+    """
+    # List of checked files and their results
+    javascript_files = []
+    # List of errors
+    errors = []
+
+
+    # Find all javascript files
+    for filename in _get_list_of_current_changed():
+        try:
+            if _is_javascript_file(filename):
+                javascript_files.append(str(filename))
+        except IOError:
+            print 'File not found (probably deleted): {}\t\tSKIPPED'.format(
+                filename)
+
+    # Do nothing if  no Javascript files changed
+    if len(javascript_files) == 0:
+        print "No Javascript file to check"
+        sys.exit(0)
+
+    # lint javascript files
+    i = 1
+    for javascript_file in javascript_files:
+
+        # Start  linting
+        print "Running jscs linting on {} (file {}/{})..\t".format(
+            javascript_file, i, len(javascript_files))
+        print ""
+        try:
+            proc = subprocess.Popen(
+                [path_to_jscs, javascript_file],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE)
+            out, _ = proc.communicate()
+            errors.append(out)
+            print out
+        except OSError:
```

Comment:
This hides any errors, which is probably going to make debugging them annoying. Maybe better to just raise the error to the developer.